src/load_order.ts

```diff
@@ -355,14 +361,25 @@ const compileDetesToGenerateLoadOrderUi: VortexWrappedDeserializeFunc = async (
     return Promise.reject(new Error(`${me}: Invalid profile or wrong game, canceling: ${jsonp(activeProfile)}`));
   }
 
-  vortexApi.log(`info`, `${me}: Compiling detes for load order UI}`);
+  vortexApi.log(`info`, `${me}: Compiling detes for load order UI`);
+
+  const deserializedLoadOrder = await pipe(
+    loadOrderPathFor(activeProfile, gameDirPath),
+    deserializeLoadOrder(vortexApi),
+  )();
+
+  // The rest of this function could and should be refactored into a pipeline
+  // to get rid of this early return
+
+  if (isLeft(deserializedLoadOrder)) {
+    vortexApi.log(`error`, `${me}: Error deserializing load order: ${deserializedLoadOrder.left.message}`);
+    showInvalidLoadOrderFileErrorDialog(vortexApi, loadOrderPathFor(activeProfile, gameDirPath));
+
+    return Promise.reject(deserializedLoadOrder.left);
+  }
```

Comment:
Will this promise rejection stop complete crashes at the very least?

Reply:
Nothing fully crashes, it's just hard to interpret, but most importantly this will prevent overwriting the file in case the decoding failure was because of a race condition.